ampersand-state.js

```diff
@@ -625,16 +626,16 @@ var dataTypes = {
             if (newVal == null) {
                 newType = typeof null;
             } else if (!isDate(newVal)) {
-                try {
-                    var dateVal = new Date(newVal).valueOf();
-                    if (isNaN(dateVal)) {
-                        // If the newVal cant be parsed, then try parseInt first
-                        dateVal = new Date(parseInt(newVal, 10)).valueOf();
-                        if (isNaN(dateVal)) throw TypeError;
-                    }
-                    newVal = dateVal;
-                    newType = 'date';
-                } catch (e) {
+                var err = null;
+                var dateVal = new Date(newVal).valueOf();
```

Comment:
iirc the original try/catch was there because it was believed there was some value for `newVal` that would make date throw an exception. However for the life of me I can't recreate that so maybe I was mistaken there.  I'm all for this refactor if it's the case that `new Date(...)`  will never throw an exception.